proxy/src/http/sql_over_http.rs

```diff
@@ -223,7 +247,13 @@ pub async fn handle(
                 }
             }
             transaction.commit().await?;
-            Ok(json!({ "results": results }))
+            Ok(json!({
+                "results": results,
+                "options": {
+                    "isolation_level": txn_isolation_level.map(|l| format!("{:?}", l)),
```

Comment:
I don't particularly trust the Debug impl. Can we instead use the value directly from the header here?

Also, do we need to return this? Maybe we can return them back as headers instead of json values?

Reply:
b/c isolation level and read only property is something that users are frequently confused about "does this really take effect??" and therefore we should let them know if this is really taking effect. returning the options in header looks good and I will do that.

Reply:
fixed in https://github.com/neondatabase/neon/pull/4830/commits/8870ed8f6d4dbe80ba3af794f4d9f9f118f7b358 and added tests in https://github.com/neondatabase/neon/pull/4830/commits/a866b3283b4a04dce48228b649a26eb8a4ce170a